static/app/views/replays/detail/replayEvents.tsx

```diff
@@ -0,0 +1,122 @@
+import {RouteComponentProps, withRouter} from 'react-router';
+import styled from '@emotion/styled';
+import first from 'lodash/first';
+
+import AsyncComponent from 'sentry/components/asyncComponent';
+import EventOrGroupTitle from 'sentry/components/eventOrGroupTitle';
+import EventMessage from 'sentry/components/events/eventMessage';
+import RRWebIntegration from 'sentry/components/events/rrwebIntegration';
+import * as Layout from 'sentry/components/layouts/thirds';
+import TagsTable from 'sentry/components/tagsTable';
+import space from 'sentry/styles/space';
+import {Event} from 'sentry/types/event';
+import EventView from 'sentry/utils/discover/eventView';
+import {getMessage} from 'sentry/utils/events';
+
+import {Replay} from '../types';
+
+type Props = AsyncComponent['props'] &
+  RouteComponentProps<{orgId: string; replayId: string}, {}> & {
+    eventSlugs: string[];
+  };
+
+type State = AsyncComponent['state'] & {
+  replayList: Replay[] | null;
+};
+
+const EventHeader = ({event}: {event: Event}) => {
+  const message = getMessage(event);
+  return (
+    <EventHeaderContainer data-test-id="event-header">
+      <TitleWrapper>
+        <EventOrGroupTitle data={event} /> {event.id}
+      </TitleWrapper>
+      {message && (
+        <MessageWrapper>
+          <EventMessage message={message} />
+        </MessageWrapper>
+      )}
+    </EventHeaderContainer>
+  );
+};
+
+class ReplayEvents extends AsyncComponent<Props, State> {
+  getEndpoints(): ReturnType<AsyncComponent['getEndpoints']> {
+    const {eventSlugs, params, location} = this.props;
+
+    return eventSlugs.map(eventSlug => {
+      const eventView = EventView.fromLocation(location);
+      const query = eventView.getEventsAPIPayload(location);
```

Comment:
We should leave this leave as-is for now, but I wonder if we will need this at all. I *think* this is needed on e.g. "list" views where we need to filter by project/env/date.

Reply:
this is all different now with the `useApi` hook. With the way the data model is now, loading loads of attachments and `sentry-replay-event` objects, pagination is something to look at, but I did find some big examples and it seems like we can get away without more complex loading for now.